tests/test_device.py

```diff
@@ -722,6 +724,48 @@ def test_outdated_API(self, monkeypatch):
             with pytest.raises(DeviceError, match="plugin requires PennyLane versions"):
                 qml.device("default.qubit", wires=0)
 
+    def test_refresh_entrypoints(self, monkeypatch):
+        """Test that new entrypoints are found by the refresh_devices function"""
+
```

Comment:
Here and in the next test too, I think it would be good for understanding the test cases to include an `assert qml.plugin_devices` before the `monkeypatch` context.

Calling `importlib.reload(qml)` within the context essentially sets `qml.plugin_devices` to `{}` and this seems to be the reason for the tear-down call of `importlib.reload(qml)` too.

Reply:
Have incorporated your comment as a comment within the test!